packages/mobx/src/types/actionannotation.ts

```diff
@@ -0,0 +1,131 @@
+import {
+    ObservableObjectAdministration,
+    createAction,
+    isAction,
+    defineProperty,
+    getDescriptor,
+    objectPrototype,
+    die,
+    isFunction,
+    Annotation,
+    recordAnnotationApplied
+} from "../internal"
+
+export function createActionAnnotation(name: string, options?: object): Annotation {
+    return {
+        annotationType_: name,
+        options_: options,
+        make_,
+        extend_
+    }
+}
+
+function make_(adm: ObservableObjectAdministration, key: PropertyKey): void {
+    let annotated = false
+    let source = adm.target_
+    // Bound action still applies normal action to prototypes,
```

Comment:
Tests check for this, but there isn't a test that would demonstrate and actual need or use case.
I've implemented it more or less like this just to be sure - I know that it's technically possible to call `super.boundAction` from non-action - async context, non-action method, perhaps constructor? - I think it would be a very edgy case, but I might be missing something as well. 
I could surely go with "lets wait until someone hits a problem", but some of these things are so suttle, that I think people often wouldn't even notice.
Or, possibly, there is stronger reason that I just forgot :D

Reply:
Ok, I think in that case I'd skip it. I think it is easier to justify to add it later if it is really a problem, compared to removing when it is in the way for someone, when it interferes with someone monkey patching it in a test suite or something obscure. If we're not sure if it adds value I prefer to do nothing, which is cheaper and no code :)

Reply:
Done